models/src/main/scala/coop/rchain/models/EqualsM.scala

```diff
@@ -101,9 +102,8 @@ object EqualM extends EqualMDerivation {
   implicit val ConnectiveEqual = gen[Connective]
   implicit def SignedEqual[A: EqualM] = new EqualM[Signed[A]] {
     override def equal[F[_]: Sync](self: Signed[A], other: Signed[A]): F[Boolean] =
-      EqualM[A]
-        .ensuring((self.sigAlgorithm, self.sig) == (other.sigAlgorithm, other.sig))
-        .equal(self.data, other.data)
+      ((self.sigAlgorithm, self.sig) == (other.sigAlgorithm, other.sig))
```

Comment:
Hye, this is still not optimal because if the first part is `false` the other does not have too be executed. Also allocation of two tuples for a method like this is wasteful. IMO this should be sth like:
```scala
if (self.sigAlgorithm == other.sigAlgorithm && self.sig == other.sig) EqualM[A].equal(self.data, other.data) else Sync[F].pure(false)
```
```